apis/apps/v1/componentdefinition_types.go

```diff
@@ -0,0 +1,1973 @@
+	// The operator waits for each replica to be updated and ready before proceeding to the next one.
+	// This ensures that only one replica is unavailable at a time during the update process.
+	SerialStrategy UpdateStrategy = "Serial"
+
+	// ParallelStrategy indicates that updates are applied simultaneously to all Pods of a Component.
+	// The replicas are updated in parallel, with the operator updating all replicas concurrently.
+	// This strategy provides the fastest update time but may lead to a period of reduced availability or
+	// capacity during the update process.
+	ParallelStrategy UpdateStrategy = "Parallel"
+
+	// BestEffortParallelStrategy indicates that the replicas are updated in parallel, with the operator making
+	// a best-effort attempt to update as many replicas as possible concurrently
+	// while maintaining the component's availability.
+	// Unlike the `Parallel` strategy, the `BestEffortParallel` strategy aims to ensure that a minimum number
+	// of replicas remain available during the update process to maintain the component's quorum and functionality.
+	//
+	// For example, consider a component with 5 replicas. To maintain the component's availability and quorum,
+	// the operator may allow a maximum of 2 replicas to be simultaneously updated. This ensures that at least
+	// 3 replicas (a quorum) remain available and functional during the update process.
+	//
+	// The `BestEffortParallel` strategy strikes a balance between update speed and component availability.
+	BestEffortParallelStrategy UpdateStrategy = "BestEffortParallel"
+)
+
+// ReplicaRole represents a role that can be assumed by a component instance.
+type ReplicaRole struct {
+	// Defines the role's identifier. It is used to set the "apps.kubeblocks.io/role" label value
+	// on the corresponding object.
+	//
+	// This field is immutable once set.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=32
+	// +kubebuilder:validation:Pattern=`^.*[^\s]+.*$`
+	Name string `json:"name"`
+
+	// Indicates whether a replica assigned this role is capable of providing services.
+	//
+	// This field is immutable once set.
+	//
+	// +kubebuilder:default=false
+	// +optional
+	Serviceable bool `json:"serviceable,omitempty"`
+
+	// Determines if a replica in this role has the authority to perform write operations.
+	// A writable replica can modify data, handle update operations.
+	//
+	// This field is immutable once set.
+	//
+	// +kubebuilder:default=false
+	// +optional
+	Writable bool `json:"writable,omitempty"`
+
+	// Specifies whether a replica with this role has voting rights.
+	// In distributed systems, this typically means the replica can participate in consensus decisions,
+	// configuration changes, or other processes that require a quorum.
+	//
+	// This field is immutable once set.
+	//
+	// +kubebuilder:default=false
+	// +optional
+	Votable bool `json:"votable,omitempty"`
```

Comment:
could be deleted

Reply:
The original name of this field was 'votable', which doesn't seem appropriate. 

For example, if we have a leader and two loggers, they are all votable, but we can't stop the leader. Only the leader is in the quorum.